modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/externalize/RelJson.java

```diff
@@ -874,6 +875,11 @@ RexNode toRex(RelInput relInput, Object o) {
                     return rexBuilder.makeNullLiteral(type);
                 }
 
+                // Avoid different representation for exact number and approx type.
+                if ((literal instanceof Long || literal instanceof Integer) && isApproximateNumeric(type)) {
+                    literal = new BigDecimal(((Number) literal).longValue());
```

Comment:
let's add comment describing what the problem is, and how this particular IF statement address it

Reply:
ok i just add one more comment line, i think that no need to describe all calcite code transformations here, also it can be changed in future.